packages/strapi-admin/services/permission.js

```diff
@@ -180,6 +189,43 @@ const cleanPermissionInDatabase = async () => {
   }
 };
 
+const ensureBoundPermissionsInDatabase = async () => {
+  if (strapi.EE) return;
+
+  const models = Object.values(strapi.contentTypes);
+  const role = await strapi.query('role', 'admin').findOne({ code: EDITOR_CODE }, []);
+
+  for (const model of models) {
+    const actions = getBoundActionsBySubject(role, model.uid);
+    const perms = await strapi.query('permission', 'admin').find(
+      {
+        subject: model.uid,
+        action_in: actions,
+        role: role.id,
+      },
+      []
+    );
+
+    if (perms.length > 0) {
```

Comment:
I would return if length === 0 to make the code less nested and because it will be the most commong usecase